### ISSUE TITLE
Sharing Regex on my website (embed regex101.com)

### ISSUE THREAD

**Issue by slhuilli1** [firasdib/Regex101#2219]
## Feature
Hi,
From my website, I share all I code for Joomla plugin. I use Regex101 for testing my regex. When my regex is OK, I integrate it into my code. But I write an article for explain to my community what I code.
I'd like to share my regex work on my website (I write article for each Joomla plugin I write), so, like some website, I suggest you a **code generator** for embedding my work from regex101 to my webpages, like an iframe, with ad of your site for example
I'd like like to share this "printscreen"
![image_2024-01-30_183342440](https://github.com/firasdib/Regex101/assets/104381156/a7916cda-5abf-4d74-8a1c-a9a41872d92d)

**Comment by hediet** [comment]
If the iframe would react to window.onmessage and send changes of the regexp back via `parent.sendmessage`, this would allow for some very nice embedding scenarios!

Since you are in control of the content of the iframe, you can still monetize the content, e.g. by showing subtle ads.

(I'm a VS Code engineer, currently exploring iframe based custom editors - this would be a super nice use-case)

**Comment by firasdib** [comment]
Right, I can see the value in this. I wonder if there are any security implications I'd have to consider?

**Comment by slhuilli1** [comment]
My proposal is not to interact from the website but just to share the regex in read-only mode. A bit like Facebook does with sharing a publication

**Comment by hediet** [comment]
> Right, I can see the value in this. I wonder if there are any security implications I'd have to consider?

The outer iframe cannot access any data/dom of the inner iframe, if they are on different domains.
However, the outer iframe can send and listen to messages of the inner iframe.

You just have to be aware of clickjacking - when an embedder loads the iframe, the user is authenticated inside that iframe.
I don't think this is a problem for regex101 though.
Maybe there could be some `https://regex101.com/embedded` even, which would be the only page where the server sends `X-Frame-Options: allow`?

> 
My proposal is not to interact from the website but just to share the regex in read-only mode. A bit like Facebook does with sharing a publication

Bidirectional editing would allow for some very interesting use-cases in code editors, such as VS Code.

**Comment by firasdib** [comment]
I'm working on a read-only mode of the main regex editor page that will be embeddable by prepending the url with `/embed`. This will hopefully make it into the upcoming version.

**Comment by firasdib** [comment]
Try out the new functionality at `https://regex101.com/embed/`

**Comment by kussmaul** [comment]
`https://regex101.com/embed/`, though the final `/` is required, which threw me at first. Thank you!

Here's a simple example with url parameters:
`<iframe src="https://regex101.com/embed/?regex=^ab*a$&testString=aa%0Aaba%0Aabba%0Aabbba" height="325px" width="400px"></iframe>`

**Comment by slhuilli1** [comment]
Great ! Great job, thanks a lot
